op-node/rollup/engine/payload_success.go

```diff
@@ -51,5 +52,42 @@ func (e *EngineController) onPayloadSuccess(ctx context.Context, ev PayloadSucce
 	err := e.tryUpdateEngineInternal(ctx)
 	if err != nil {
 		e.log.Error("Failed to update engine", "error", err)
+	} else {
+		updateEngineFinish := time.Now()
+		e.logBlockProcessingMetrics(updateEngineFinish, ev)
 	}
 }
+
+func (e *EngineController) logBlockProcessingMetrics(updateEngineFinish time.Time, ev PayloadSuccessEvent) {
+	// Protect against nil pointer dereferences
+	if ev.Envelope == nil || ev.Envelope.ExecutionPayload == nil {
+		e.log.Debug("Envelope.ExecutionPayload not found, skipping block processing metrics")
+		return
+	}
+
+	buildTime := ev.InsertStarted.Sub(ev.BuildStarted)
+	insertTime := updateEngineFinish.Sub(ev.InsertStarted)
+
+	var totalTime time.Duration
+	if !ev.BuildStarted.IsZero() {
+		totalTime = updateEngineFinish.Sub(ev.BuildStarted)
+	} else {
+		totalTime = insertTime
+	}
+
+	// Protect against divide-by-zero
+	var mgasps float64
+	if totalTime > 0 {
+		mgasps = float64(ev.Envelope.ExecutionPayload.GasUsed) * 1000 / float64(totalTime)
+	}
```

Comment:
why not just not include `mgasps`? `mgas` and `total_time` are both available and we can calculate it in post if needed.

Reply:
Yes good point. Was thinking we may create a prometheus metric for `mgasps`, but will add it back later if we do indeed want a metric for that.

Reply:
mgasps measured in this way is a little misleading anyway, since it isn't real time, only execution time. It could show 100 gas in 0.2 seconds for what *appears to be* 500 gasps. But if that's the only block being published for 2 seconds, the *realtime* gasps is actually just 50.